students/DennisLee/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/env python3
+
+import os
+import mailroom as m
+import pytest
+
+
+choices = {  # Dict of command text and functions in the main menu
+'1': {'option': 'Send a thank you', 'function': m.send_thank_you},
+'2': {'option': 'Create a report', 'function': m.create_a_report},
+'3': {'option': 'Send letters to everyone', 'function': m.send_all_letters},
+'4': {'option': 'Quit', 'function': m.exit_screen}
+}
+
+# Dict of functions to show donor list or quit when sending a thank you
+alt_choices = {  
+        '': {'function': m.exit_screen},
+        'quit': {'function': m.exit_screen},
+        'list': {'function': m.print_donor_list}
+}
+
+
+def test_manage_donors_1():
+    # An I/O error occurs because the function prompts for menu choice
+    with pytest.raises(IOError):
+        m.manage_donors()
+
+# Simulate and test function calls that occur in the code 
+# after the input() function that causes the I/O error
+# There aren't menu options for 0', '5', 'a', & '', so they return False
+# '1' and '3' lead to user prompts, so catch those
+# '2' and '4' go to functions w/o user prompts, so they return True
+def test_call_menu_function_1():
+    assert m.call_menu_function(
+            choices, '0', m.respond_to_bad_main_menu_choice) == False
```

Comment:
Generally in Python/Pep8, you want to use the `is` operator for comparison to `False`, or in most cases just do `assert not function_that_returns_a_boolean()`.

Reply:
If a function can return `None`, wouldn't `not None` evaluate to `True`? That's why I haven't been using the simpler expression.

Reply:
There are definitely some cases where it makes sense to go against the conventions and you'll see boolean comparisons that use the `==` operator in plenty of Python code. I just wanted to mention this, as a general Python style consideration. 

Here's a stack overflow post that speaks to the topic and links to the relevant Pep8 documentation if you want to learn more: https://softwareengineering.stackexchange.com/a/322394